ui/apps/platform/src/Components/CompoundSearchFilter/components/CheckboxSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement } from 'react';
+import {
+    Select,
+    SelectOption,
+    SelectList,
+    MenuToggle,
+    MenuToggleElement,
+    Badge,
+} from '@patternfly/react-core';
+
+type CheckboxSelectProps = {
+    selection: string[];
+    onChange: (value: string[]) => void;
+    children: ReactElement<typeof SelectOption> | ReactElement<typeof SelectOption>[];
+    isDisabled?: boolean;
+    ariaLabelMenu?: string;
+    toggleLabel?: string;
+};
+
+function CheckboxSelect({
+    selection,
+    onChange,
+    children,
+    isDisabled = false,
+    ariaLabelMenu,
+    toggleLabel,
+}: CheckboxSelectProps) {
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const onSelect = (
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        value: string | number | undefined
+    ) => {
+        if (selection.includes(value as string)) {
+            onChange(selection.filter((id) => id !== value));
+        } else {
+            onChange([...selection, value as string]);
+        }
```

Comment:
What do you think about:
```suggestion
        if (selection.includes(String(value))) {
            onChange(selection.filter((id) => id !== value));
        } else {
            onChange([...selection, String(value)]);
        }
```
both cases would probably lead to invalid data, but would one be better than the other?
`['test', '', 2, undefined]`
versus
`['test', '', '2', 'undefined']`

Would it be better to test for a string value and return early? Or is neither situation something that will happen in practice?

Reply:
Hopefully, the developer will use the correct `SelectOption`s but I agree that they could use a combination of string values, number values, and undefined (if they intended to). I think I'll need to look into this further. For now, we can go with your approach